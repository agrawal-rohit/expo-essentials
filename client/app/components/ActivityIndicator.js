import React, { useEffect } from "react";
import { View, StyleSheet } from "react-native";
import { Spinner } from "@ui-kitten/components";

function ActivityIndicator({ visible = false, ...props }) {

  if(!visible) return null

  return (
    <View style={styles.overlay}>
      <View style={styles.container}>
        <Spinner size="large" {...props} />
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  overlay: {
    width: "100%",
    height: "100%",
    backgroundColor: "white",
    position: "absolute",
    opacity: 0.8,
    zIndex: 1,
  },
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
});

export default ActivityIndicator;
