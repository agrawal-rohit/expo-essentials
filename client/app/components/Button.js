import React from 'react';
import { View, StyleSheet } from 'react-native';
import { Button, Spinner } from '@ui-kitten/components';
import SubHeading from './SubHeading';

const styles = StyleSheet.create({
  buttonTextStyle: {
    fontSize: 14,
    textTransform: 'uppercase',
    color: 'white',
  },
  indicator: {
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default function CustomButton({
  children,
  color = null,
  loading = false,
  style = {},
  ...props
}) {
  const LoadingIndicator = () => (
    <View style={styles.indicator}>
      <Spinner status="basic" size="small" />
    </View>
  );

  const ButtonText = ({ children, loading }) => {
    if (loading) return null;
    return <SubHeading style={styles.buttonTextStyle}>{children}</SubHeading>;
  };

  // Custom color provided
  if (color) {
    return (
      <Button
        children={ButtonText({ children, loading })}
        accessoryLeft={loading ? LoadingIndicator : null}
        style={{ ...style, backgroundColor: color, borderColor: color }}
        {...props}
      />
    );
  }

  return (
    <Button
      children={ButtonText({ children, loading })}
      accessoryLeft={loading ? LoadingIndicator : null}
      style={style}
      {...props}
    />
  );
}
