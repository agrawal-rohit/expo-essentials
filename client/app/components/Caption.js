import React from 'react';
import { Text } from '@ui-kitten/components';

export default function Heading({
  children,
  fontSize = 15,
  style,
  ...props
}) {
  return (
    <Text
      category="c1"
      style={{
        fontSize,
        fontFamily: 'Jost-Regular',
        ...style,
      }}
      {...props}
    >
      {children}
    </Text>
  );
}
