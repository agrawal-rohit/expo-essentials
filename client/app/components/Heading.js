import React from 'react';
import { Text } from '@ui-kitten/components';

export default function Heading({
  children,
  fontSize = 28,
  style = {},
  ...props
}) {
  return (
    <Text
      category="h1"
      style={{
        fontSize,
        fontFamily: 'Jost-Bold',
        ...style,
      }}
      {...props}
    >
      {children}
    </Text>
  );
}
