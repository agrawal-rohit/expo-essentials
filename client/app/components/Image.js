import React, { useContext } from "react";
import { Image } from "react-native";
import { Image as CacheImage } from "react-native-expo-image-cache";
import { ThemeContext } from "../contexts/theme-context";

export default function CustomImage({
  source = null,
  imageUri = null,
  preview = null,
  cache = true,
  ...props
}) {
  const { theme } = useContext(ThemeContext);

  if (cache) {
    return <CacheImage uri={imageUri} preview={preview} tint={theme} {...props} />;
  }

  if (imageUri) {
    return <Image source={{ uri: imageUri }} {...props} />;
  }

  return <Image source={source} {...props} />;
}
