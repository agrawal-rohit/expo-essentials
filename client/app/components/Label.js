import React from 'react';
import { Text } from '@ui-kitten/components';

export default function Heading({
  children,
  fontSize = 12,
  fontWeight = '800',
  style,
  ...props
}) {
  return (
    <Text
      category="label"
      style={{
        fontSize,
        fontWeight,
        fontFamily: 'Jost-Bold',
        ...style,
      }}
      {...props}
    >
      {children}
    </Text>
  );
}
