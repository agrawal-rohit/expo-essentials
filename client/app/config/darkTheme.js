import colors from './colors';

export default {
  
  ...colors,

  "text-basic-color": "$color-basic-100",
  
  "color-primary-default": "$color-primary-500",
  "color-primary-hover": "$color-primary-400",
  "color-primary-focus": "$color-primary-600",
  "color-primary-active": "$color-primary-600",
  "color-primary-disabled": "$color-basic-transparent-300",
  "color-primary-default-border": "$color-primary-default",
  "color-primary-hover-border": "$color-primary-hover",
  "color-primary-focus-border": "$color-primary-700",
  "color-primary-active-border": "$color-primary-active",
  "color-primary-disabled-border": "$color-primary-disabled",

  "background-basic-color-1": "$color-basic-800",
  "background-basic-color-2": "$color-basic-900",
  "background-basic-color-3": "$color-basic-1000",
  "background-basic-color-4": "$color-basic-1100",

  "notification-success": "$color-success-600",
  "notification-error": "$color-danger-600"
}
