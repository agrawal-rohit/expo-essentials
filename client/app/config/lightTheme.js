import colors from './colors';

export default {
  
  ...colors,

  "text-basic-color": "$color-basic-800",
  
  "color-primary-default": "$color-primary-500",
  "color-primary-hover": "$color-primary-400",
  "color-primary-focus": "$color-primary-600",
  "color-primary-active": "$color-primary-600",
  "color-primary-disabled": "$color-basic-transparent-300",
  "color-primary-default-border": "$color-primary-default",
  "color-primary-hover-border": "$color-primary-hover",
  "color-primary-focus-border": "$color-primary-700",
  "color-primary-active-border": "$color-primary-active",
  "color-primary-disabled-border": "$color-primary-disabled",

  "background-basic-color-1": "$color-basic-100",
  "background-basic-color-2": "$color-basic-200",
  "background-basic-color-3": "$color-basic-300",
  "background-basic-color-4": "$color-basic-400",

  "notification-success": "$color-success-600",
  "notification-error": "$color-danger-600"
}
