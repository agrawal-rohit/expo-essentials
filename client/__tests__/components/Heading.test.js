/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';

import { ApplicationProvider } from '@ui-kitten/components';
import * as eva from '@eva-design/eva';

import Heading from '../../app/components/Heading';
import lightTheme from '../../app/config/lightTheme';
import darkTheme from '../../app/config/darkTheme';

describe('<Heading />', () => {
  it('has 1 child', () => {
    const tree = renderer
      .create(
        <ApplicationProvider {...eva} theme={eva.light}>
          <Heading>Test Heading</Heading>
        </ApplicationProvider>,
      )
      .toJSON();
    expect(tree.children.length).toBe(1);
  });

  it('works correctly in light theme', () => {
    const tree = renderer
      .create(
        <ApplicationProvider {...eva} theme={{ ...eva.light, ...lightTheme }}>
          <Heading>Test Heading</Heading>
        </ApplicationProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('works correctly in dark theme', () => {
    const tree = renderer
      .create(
        <ApplicationProvider {...eva} theme={{ ...eva.dark, ...darkTheme }}>
          <Heading>Test Heading</Heading>
        </ApplicationProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
