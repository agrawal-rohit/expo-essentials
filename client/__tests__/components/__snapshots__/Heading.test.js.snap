// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> works correctly in dark theme 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    appearance="default"
    category="h1"
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "System",
          "fontSize": 36,
          "fontWeight": "800",
        },
        Object {
          "fontFamily": "Jost-Bold",
          "fontSize": 28,
        },
      ]
    }
  >
    Test Heading
  </Text>
</View>
`;

exports[`<Heading /> works correctly in light theme 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    appearance="default"
    category="h1"
    style={
      Array [
        Object {
          "color": "#222B45",
          "fontFamily": "System",
          "fontSize": 36,
          "fontWeight": "800",
        },
        Object {
          "fontFamily": "Jost-Bold",
          "fontSize": 28,
        },
      ]
    }
  >
    Test Heading
  </Text>
</View>
`;
